test/TestAlternativesPerformance.cpp

```diff
@@ -0,0 +1,268 @@
+#include "../src/AlternativesPerformance.h"
+#include "../src/Criteria.h"
+#include "../src/Performance.h"
+#include "../src/PerformanceTable.h"
+#include "gtest/gtest.h"
+#include <sstream>
+#include <utility>
+
+TEST(TestAlternativesPerformance, TestConstructorBaseConstructor) {
+  Criteria crit = Criteria(2, "a");
+
+  AlternativesPerformance alt_perf = AlternativesPerformance(2, crit, "test");
+  std::ostringstream os;
+  os << alt_perf;
+  EXPECT_EQ(
+      os.str(),
+      "AlternativesPerformance( PerformanceTable[ Performance: Perf( name : "
+      "test0, crit : a0, value : 0 ) Perf( name : test0, crit : a1, value : 0 "
+      ") | Performance: Perf( name : test1, crit : a0, value : 0 ) Perf( name "
+      ": test1, crit : a1, value : 0 ) | ], AlternativesAssignment{ test0->"
+      " test1-> }");
+  EXPECT_EQ(alt_perf.getMode(), "alt");
+
+  std::map<std::string, std::string> map =
+      std::map<std::string, std::string>{{"test0", "cat0"}, {"test1", "cat1"}};
+  AlternativesPerformance alt_perf2 =
+      AlternativesPerformance(2, crit, "test", map);
+  std::ostringstream os2;
+  os2 << alt_perf2;
+  EXPECT_EQ(
+      os2.str(),
+      "AlternativesPerformance( PerformanceTable[ Performance: Perf( name : "
+      "test0, crit : a0, value : 0 ) Perf( name : test0, crit : a1, value : 0 "
+      ") | Performance: Perf( name : test1, crit : a0, value : 0 ) Perf( name "
+      ": test1, crit : a1, value : 0 ) | ], AlternativesAssignment{ test0->cat0"
+      " test1->cat1 }");
+  EXPECT_EQ(alt_perf2.getMode(), "alt");
```

Comment:
Names tend to confuse : name of a performance should be 'a1' (alternative1) or 'b1' (profile1) if we consider notations from MDCA.
Criteria could be : crit1, etc....

Reply:
Not sure we followed this notation for the previous tests, but indeed it will be more understandable if we start using it ! Especially if we turn our tests in documentation later :)

Reply:
I used the same notation as Thomas to be consistent lol :) but right this is confusing sometimes, I'll correct it